connect/runtime/src/test/java/org/apache/kafka/connect/integration/SessionedProtocolIntegrationTest.java

```diff
@@ -115,11 +112,11 @@ public void ensureInternalEndpointIsSecured() throws Throwable {
         // Try again, but with an invalid signature
         log.info(
             "Making a POST request to the {} endpoint with no connector started and an invalid signature header; "
-                + "expecting 403 error response",
+                + "expecting 503 error response",
             connectorTasksEndpoint
         );
         assertEquals(
-            FORBIDDEN.getStatusCode(),
+            SERVICE_UNAVAILABLE.getStatusCode(),
```

Comment:
Could you share the details with me?

Reply:
Here, if the connector is not started, there is no `sessionKey`.

https://github.com/apache/kafka/blob/05df10449eb9c95fe6d6055b302c84686be8058d/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java#L2748-L2752

Reply:
This was added in https://github.com/apache/kafka/pull/11783.

Reply:
CI can't pass. Not sure whether there is some connector forgot to close. The error is:

```
java.lang.AssertionError: expected:<503> but was:<403>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.apache.kafka.connect.integration.SessionedProtocolIntegrationTest.ensureInternalEndpointIsSecured(SessionedProtocolIntegrationTest.java:118)
```

Reply:
`503` is a temporary error, because the worker is still in starting. Could you please add a wait to check the 503 first ?

Reply:
Added it and `ensureInternalEndpointIsSecured` can pass in CI environment. Thanks.